src/payment/strategies/amazon-pay/amazon-pay-payment-strategy.spec.ts

```diff
@@ -468,7 +470,26 @@ describe('AmazonPayPaymentStrategy', () => {
         }
     });
 
-    describe('When 3ds is enabled', () => {
+    describe( '#initialize', () => {
+        it('throws PaymentMethodNotAllowedError', () => {
+            const remoteCheckout = {
+                referenceId: 'referenceId',
+                billing: {
+                    address: {},
+                    paymentMethodNotAllowed: true,
+                },
+            };
+
+            jest.spyOn(store, 'getState');
+            jest.spyOn(store.getState().remoteCheckout, 'getCheckout')
+                .mockReturnValue(remoteCheckout);
+
+            expect(strategy.initialize({ methodId: paymentMethod.id, amazon: { container: 'wallet' } }))
+                .resolves.toThrow(PaymentMethodNotAllowedError);
+        });
+    });
+
+    describe('#execute When 3ds is enabled', () => {
```

Comment:
Your description of a `describe` needs to be each public method, in this case `execute`, `when 3ds is enabled` is a more detailed tests, so I suggest to remove from here and add it in each test that applies 🍹